resource/resource.go

```diff
@@ -0,0 +1,14 @@
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sync"
+)
+
+// needs to be accessible by both project controllers and mgmt controllers
+var (
+	CleanerFuncMap = struct {
```

Comment:
change the name of this to be `ClusterScopedFinalizerCleanup

Also, generally, putting the type (ie "Map") in the name of  a variable is unnecessary

Reply:
add a comment explaining what this variable is used for

Reply:
This is technically more than a list of cluster scoped finalizers. This is a set of all rancher generated types.